nemo/collections/tts/modules/common.py

```diff
@@ -192,7 +192,7 @@ def __init__(self, c):
         # Ensure determinant is 1.0 not -1.0
         if torch.det(W) < 0:
             W[:, 0] = -1 * W[:, 0]
-        p, lower, upper = torch.lu_unpack(*torch.lu(W))
+        p, lower, upper = torch.lu_unpack(*torch.linalg.lu(W))
```

Comment:
this change is **not correct**. Per the statement in pytorch 2.0 doc, "[torch.linalg.lu()](https://pytorch.org/docs/stable/generated/torch.linalg.lu.html#torch.linalg.lu) computes the LU decomposition with partial pivoting of a possibly non-square matrix. It is a composition of [lu_factor()](https://pytorch.org/docs/stable/generated/torch.linalg.lu_factor.html#torch.linalg.lu_factor) and [torch.lu_unpack()],(https://pytorch.org/docs/stable/generated/torch.lu_unpack.html#torch.lu_unpack).", while "LU, pivots, info = torch.lu(A, compute_pivots) should be replaced with `LU, pivots = torch.linalg.lu_factor(A, compute_pivots)`". So the correct change should be,
```suggestion
        p, lower, upper = torch.linalg.lu(W)
```

See my validation code snippet,
```
In [25]: a_lu
Out[25]:
tensor([[[-8.3251e-01, -2.0748e-01, -1.6207e-03],
         [-3.0119e-01,  2.1512e+00,  1.7865e-01],
         [-5.4838e-01,  9.3401e-02, -2.2064e-01]],

        [[ 8.0517e-01, -1.3429e+00,  6.1186e-01],
         [ 9.6715e-01,  2.8609e+00, -1.9533e-01],
         [-5.8811e-01, -3.4345e-01,  6.3169e-01]]])

In [26]: pivots
Out[26]:
tensor([[2, 2, 3],
        [2, 2, 3]], dtype=torch.int32)

In [27]: p, lower, upper = torch.lu_unpack(*torch.lu(a))

In [28]: p
Out[28]:
tensor([[[0., 1., 0.],
         [1., 0., 0.],
         [0., 0., 1.]],

        [[0., 1., 0.],
         [1., 0., 0.],
         [0., 0., 1.]]])

In [29]: lower
Out[29]:
tensor([[[ 1.0000,  0.0000,  0.0000],
         [-0.3012,  1.0000,  0.0000],
         [-0.5484,  0.0934,  1.0000]],

        [[ 1.0000,  0.0000,  0.0000],
         [ 0.9671,  1.0000,  0.0000],
         [-0.5881, -0.3434,  1.0000]]])

In [30]: upper
Out[30]:
tensor([[[-8.3251e-01, -2.0748e-01, -1.6207e-03],
         [ 0.0000e+00,  2.1512e+00,  1.7865e-01],
         [ 0.0000e+00,  0.0000e+00, -2.2064e-01]],

        [[ 8.0517e-01, -1.3429e+00,  6.1186e-01],
         [ 0.0000e+00,  2.8609e+00, -1.9533e-01],
         [ 0.0000e+00,  0.0000e+00,  6.3169e-01]]])

In [31]: p, lower, upper = torch.lu_unpack(*torch.linalg.lu(a))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[31], line 1
----> 1 p, lower, upper = torch.lu_unpack(*torch.linalg.lu(a))

TypeError: lu_unpack(): argument 'unpack_data' (position 3) must be bool, not Tensor

In [32]: p, lower, upper = torch.linalg.lu(a)

In [33]: p
Out[33]:
tensor([[[0., 1., 0.],
         [1., 0., 0.],
         [0., 0., 1.]],

        [[0., 1., 0.],
         [1., 0., 0.],
         [0., 0., 1.]]])

In [34]: lower
Out[34]:
tensor([[[ 1.0000,  0.0000,  0.0000],
         [-0.3012,  1.0000,  0.0000],
         [-0.5484,  0.0934,  1.0000]],

        [[ 1.0000,  0.0000,  0.0000],
         [ 0.9671,  1.0000,  0.0000],
         [-0.5881, -0.3434,  1.0000]]])

In [35]: upper
Out[35]:
tensor([[[-8.3251e-01, -2.0748e-01, -1.6207e-03],
         [ 0.0000e+00,  2.1512e+00,  1.7865e-01],
         [ 0.0000e+00,  0.0000e+00, -2.2064e-01]],

        [[ 8.0517e-01, -1.3429e+00,  6.1186e-01],
         [ 0.0000e+00,  2.8609e+00, -1.9533e-01],
         [ 0.0000e+00,  0.0000e+00,  6.3169e-01]]])
```